File: php/205-isomorphic-strings.php
<?php

class Solution 
{
    /**
     * @param String $s
     * @param String $t
     * @return Boolean
     */
    public function isIsomorphic($s, $t) 
    {
        $map = [];
        $len = strlen($s);
        for ($i=0; $i<$len; $i++) {
            $sc = $s[$i];
            $tc = $t[$i];
            //make sure different key correspond different value
            if (!isset($map[$sc]) && !in_array($tc, $map)) {
                $map[$sc] = $tc;
            }
            if ($map[$sc] != $tc) {
                return false;
            }
        }
        return true;
    }
}

File: go/205-isomorphic-strings.go
package main

func isIsomorphic(s string, t string) bool {
	ms := make(map[string]string)
	mt := make(map[string]string)
	lens := len(s)
	lent := len(t)
	if lens != lent {
		return false
	}
	for i := 0; i < lens; i++ {
		si := string(s[i])
		ti := string(t[i])
		sv, sexist := ms[si]
		tv, texist := mt[ti]
		if !sexist && !texist {
			ms[si] = ti
			mt[ti] = si
		}
		if sexist && sv != ti {
			return false
		}
		if texist && tv != si {
			return false
		}
	}
	return true
}
